Dockerfile

```diff
@@ -1,19 +1,28 @@
-FROM ubuntu:bionic
-MAINTAINER Zach Halpern <zahalpern+github@gmail.com>
+FROM ubuntu:24.04
+
+ARG DEBIAN_FRONTEND=noninteractive
 
 RUN apt-get update && apt-get install -y\
-  build-essential\
-  cmake\
-  git\
-  libprotobuf-dev\
-  libqt5sql5-mysql\
-  libmysqlclient-dev\
-  libqt5websockets5-dev\
-  protobuf-compiler\
-  qt5-default\
-  qtbase5-dev\
-  qttools5-dev-tools\
-  qttools5-dev
+  build-essential \
+        ccache \
+        clang-format \
+        cmake \
+        file \
+        g++ \
+        git \
+        libgl-dev \
+        liblzma-dev \
```

Comment:
many of these dependencies are only relevant to our ci (I guess you copied them from there)
for compiling the server none of the gui things are needed either.
remove: ccache clang-format liblzma-dev libgl-dev libqt6multimedia6 qt6-svg-dev qt6-l10n-tools qt6-multimedia-dev